### ISSUE TITLE
Barfs when using an edited bundle.yaml for machine placement with MAAS

### ISSUE THREAD

**Issue by pmatulis** [ubuntu/conjure-up#211]
```
$ dpkg-query -W juju maas conjure-up
conjure-up      2.0.0.4-0~460~ubuntu16.04.1
juju    2.0-beta11-0ubuntu1~16.04.1~juju1
maas    2.0.0~rc1+bzr5143-0ubuntu1~xenial1
$ cd /home/ubuntu
$ git clone https://github.com/battlemidget/openstack-novalxd
$ vi openstack-novalxd/bundle.yaml
$ pastebinit openstack-novalxd/bundle.yaml
http://paste.ubuntu.com/18561160/
$ conjure-up /home/ubuntu/openstack-novalxd
```

After choosing 'maas', entering IP address and API key, and then 'Add credential', I get a stacktrace but a MAAS node begins to deploy.

MAAS server address and key entered:
192.168.1.5
hAhstwyBAr3BL9sDd7:7qMT8brmrYDf5ge2c4:xYY38zTTxBA2z6vASJc33SMtzfWch2jk

I hit Ctrl-c .

The entire screen output is here:
http://paste.ubuntu.com/18562389/

There is nothing in the 3 files under /var/log/conjure-up.

**Comment by mikemccracken** [comment]
@pmatulis hi, thanks for trying this out. it looks like the 'machines' section of your yaml is not complete - the machine numbers are keys to a dictionary, and each value needs to be a dictionary also.
So instead of this:
```yaml
machines:
  "0" 
  "1" 
  "2" 
  "3" 
  "4" 
  "5" 
  "6" 
  "7" 
```

do this:
```yaml
machines:
  0:
    series: xenial
  1:
    series: xenial
  2:
    series: xenial
```
...etc

**Comment by pmatulis** [comment]
Thank you. This got me going again.

**Comment by pmatulis** [comment]
Now this:
http://paste.ubuntu.com/18570893/

**Comment by pmatulis** [comment]
After I Ctrl-c I have this on my screen:
http://paste.ubuntu.com/18571405/

**Comment by mikemccracken** [comment]
@pmatulis can you paste the output of 'juju controllers ; juju models' ?
that error is from trying to run 'juju status', and if there is a controller but no current model, it'll fail.

the easiest way to start over fresh is to destroy the controller, assuming it's not something you need to keep around, and let conjure-up recreate one.

**Comment by pmatulis** [comment]
At this time there is no controller. I'm trying again.

**Comment by pmatulis** [comment]
Was going well, and it's still going ("Deploying application: mysql"), but I got this onscreen:
http://paste.ubuntu.com/18573090/

**Comment by pmatulis** [comment]
Only one node is "Deployed" - no other activity on MAAS side (i.e. no "Deploying").

**Comment by battlemidget** [comment]
> Was going well, and it's still going ("Deploying application: mysql"), but I got this onscreen:
> http://paste.ubuntu.com/18573090/

Looks like another issue with Juju API changes. Need to verify what Juju Beta11 and onwards accepts for the placement type during a request.

**Comment by pmatulis** [comment]
bundle.yaml:
http://paste.ubuntu.com/18574442/

**Comment by pmatulis** [comment]
Guys, interesting stuff after I Ctrl-c:
http://paste.ubuntu.com/18574692/

**Comment by mikemccracken** [comment]
ok, nothing obvious in that bundle.yaml. I thikn Adam's right about the
juju api issues. We haven't exercised placement directives yet, so it might
need some different syntax

**Comment by mikemccracken** [comment]
I think what happened there is that it's still trying to do the relations,
which fails with that message when one endpoint of the relation doesn't
exist because it wasn't deployed correctly.

**Comment by mikemccracken** [comment]
@pmatulis, after #228 is merged, conjure-up should successfully deploy bundles with placement directives in them, provided that their machines dict is correct and has enough machines for all the placement directives in the bundle - the machines will be created in the new model with ID numbers starting at 0.

**Comment by battlemidget** [comment]
Should be fixed in 2.0.0.5
